src/main/resources/templates/fragments/exam-number-input.html

```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+<body>
+<div th:fragment="header">
+    <form action="#" th:action="@{/validate}" th:object="${examNumberForm}" method="post">
+        <label for="examNumber">Prüfungsnummer eingeben*:</label>
+        <input type="text" name="examNumber" id="examNumber" pattern="[0-9]{5}"
```

Comment:
Das Pattern kann an dieser Stelle weggelassen werden, weil dieses auch  in der Modellklasse ExamNumberForm definiert ist. Im Controller kann dann geprüft werden, ob die (client-seitige) Validierung einen Fehler hat oder nicht (siehe BindingResult#hasErrors() ).

Reply:
Aktuell validieren wir an drei Stellen: client-seitig anhand des Patterns, server-seitig mit der Methode viewValidatePage(@Valid ...) und schliesslich auch im Validation Service. Gemäss Beispiel https://spring.io/guides/gs/validating-form-input würde ich im Moment auf die client-seitige Validierung verzichten, solange wir nicht Scripts im client verwenden (Validierung, enable / disable Submit Button).